### PR TITLE
Added commands to antharun to list LHComponents, LHPlates and LHPolicies 

### PR DESCRIPTION
**Pull request by gogogadgetbioprocess**
Could expand to include additional flags to return properties of specific plates and policies

### CODE REVIEW

File: antha/anthalib/wtype/lhplate.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (lhp LHPlate) Name() string {
 func (lhp LHPlate) String() string {
 	return fmt.Sprintf(
 		`LHPlate {
-	ID          : %s,
+	ID          : %s,
```
Comment (ddn0) [inline antha/anthalib/wtype/lhplate.go]: gofmt please

File: cmd/antharun/cmd/plates.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	Use:   "lhplates",
+	Short: "List available antha lhplates",
+	RunE:  lhPlates,
+}
+
+func lhPlates(cmd *cobra.Command, args []string) error {
+	viper.BindPFlags(cmd.Flags())
+
+	cs := factory.GetPlateList()
+	/*if err != nil {
```
Comment (ddn0) [inline cmd/antharun/cmd/plates.go]: delete

File: cmd/antharun/cmd/plates.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	case jsonOutput:
+		if bs, err := json.Marshal(cs); err != nil {
+			return err
+		} else {
+			_, err = fmt.Println(string(bs))
+			return err
+		}
+	default:
+
+		prettystrings := make([]string, 0)
```
Comment (ddn0) [inline cmd/antharun/cmd/plates.go]: var prettyStrings []string

File: microArch/driver/liquidhandling/lhpolicy.go [hunk 4]
```diff
@@ -50,6 +50,18 @@ func LoadLHPoliciesFrom(filename string) *LHPolicyRuleSet {
 // this structure defines parameters
 type LHPolicy map[string]interface{}
 
+func (lhp LHPolicy) List() (namevaluepairs []string, names []string, values []string) {
+	namevaluepairs = make([]string, 0)
```
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: var nameValuePairs []string

File: microArch/driver/liquidhandling/lhpolicy.go [hunk 5]
```diff
@@ -50,6 +50,18 @@ func LoadLHPoliciesFrom(filename string) *LHPolicyRuleSet {
...
+	names = make([]string, 0)
```
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: var names []string

File: microArch/driver/liquidhandling/lhpolicy.go [hunk 6]
```diff
@@ -50,6 +50,18 @@ func LoadLHPoliciesFrom(filename string) *LHPolicyRuleSet {
...
+	values = make([]string, 0)
```
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: var values []string

File: microArch/driver/liquidhandling/lhpolicy.go (hunk 5 above)
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: ditto

File: microArch/driver/liquidhandling/lhpolicy.go (hunk 6 above)
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: ditto

File: microArch/driver/liquidhandling/lhpolicy.go [hunk 7]
```diff
@@ -50,6 +50,18 @@ func LoadLHPoliciesFrom(filename string) *LHPolicyRuleSet {
 // this structure defines parameters
 type LHPolicy map[string]interface{}
 
+func (lhp LHPolicy) List() (namevaluepairs []string, names []string, values []string) {
```
Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: func (lhp LHPolicy) List() (nameValuePairs, names, values []string)

Comment (ddn0) [inline microArch/driver/liquidhandling/lhpolicy.go]: In general, I'd prefer not having List() at all. A LHPolicy is a map already. It is just as easy for clients to unpack it and use it however they want.

Comment (gogogadgetbioprocess) [inline microArch/driver/liquidhandling/lhpolicy.go]: I've found myself wanting such a List function more than once so would prefer to keep

File: cmd/antharun/cmd/lhcomponents.go [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+	RunE:  lhComponents,
+}
+
+func lhComponents(cmd *cobra.Command, args []string) error {
+	viper.BindPFlags(cmd.Flags())
+
+	cs := factory.GetComponentList()
+	/*if err != nil {
+		return err
+	}*/
```
Comment (ddn0) [inline cmd/antharun/cmd/lhcomponents.go]: delete

File: cmd/antharun/cmd/lhcomponents.go [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+	case jsonOutput:
+		if bs, err := json.Marshal(cs); err != nil {
+			return err
+		} else {
+			_, err = fmt.Println(string(bs))
+			return err
+		}
+	default:
+
+		prettystrings := make([]string, 0)
```
Comment (ddn0) [inline cmd/antharun/cmd/lhcomponents.go]: var prettyStrings []string

File: cmd/antharun/cmd/lhpolicies.go [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+	case jsonOutput:
+		if bs, err := json.Marshal(policymap); err != nil {
+			return err
+		} else {
+			_, err = fmt.Println(string(bs))
+			return err
+		}
+	default:
+
+		prettystrings := make([]string, 0)
```
Comment (ddn0) [inline cmd/antharun/cmd/lhpolicies.go]: var prettyStrings []string

File: cmd/antharun/cmd/plates.go [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+			return err
+		} else {
+			_, err = fmt.Println(string(bs))
+			return err
+		}
+	default:
+
+		var prettystrings []string
+
+		prettystrings = make([]string, 0)
```
Comment (ddn0) [inline cmd/antharun/cmd/plates.go]: remove

**Pull request merged**